binding.cc

```diff
@@ -1412,87 +1455,22 @@ NAPI_METHOD(iterator_seek) {
     }
   }
 
-  NAPI_RETURN_UNDEFINED();
-}
-
-/**
- * Worker class for ending an iterator
- */
-struct EndWorker final : public BaseWorker {
```

Comment:
I like the idea of ending on GC, but it will be nice if we keep the ability to manually end an iterator as well.

One may not want to fully rely on GC on a busy server if there is a clean way to eagerly release an iterator (and the underlying snapshot and resources).

Reply:
I prefer either/or. Explicit resource management, or GC. Supporting both is not impossible (the `EndWorker` could do the same for a single iterator as what the `CloseWorker` does for all iterators now), but it raises the question of what we recommend as best practice - and whether modules that wrap iterators should end explicitly or not.

Would be nice to do some real world benchmarks.

Reply:
👍 for benchmarks. I thought ending on GC was purely for better promise support as @chjj mentioned in https://github.com/Level/leveldown/pull/597#issuecomment-472756492.

Reply:
I'm not sure _what_ to benchmark. Ending on GC will have some cost, but what cost is acceptable?

Reply:
One of my private project use iterators for prefix searching, with lots of concurrent `seek` `next` and (early) `end`. I would like to benchmark the new GC-version with my project (especially on write intensive workloads), but I'm pretty busy for the next few days...

Shall we release **v5.0.0** first and work on this PR later?

Reply:
Already released it :)

Reply:
My main concern is about the underlying snapshot. I remember that holding snapshots might affect compression of new keys after the ** SequenceNumber**, so LevelDB suggests to release snapshots as soon as possible.

Reply:
I'll assign you to this PR, no rush.